datafusion/sql/src/unparser/plan.rs

```diff
@@ -860,8 +877,12 @@ impl Unparser<'_> {
         query: &mut Option<QueryBuilder>,
         select: &mut SelectBuilder,
         relation: &mut RelationBuilder,
+        alias: &str,
     ) -> Result<()> {
         let mut unnest_relation = UnnestRelationBuilder::default();
+
+        unnest_relation.alias(Some(self.new_table_alias(alias.to_string(), vec![])));
+
```

Comment:
@goldmedal FYI, this is another change to unnest - I want to always do alias to make behaviour more consistent

Reply:
I don't get why this way can make behavior more consistent 🤔. Could you explain more?

Reply:
BTW, I'm considering how to fix the issue mentioned in (https://github.com/apache/datafusion/pull/15090#discussion_r1989018918). I think add an alias for the unnest may be a good idea. 

I think it should like
```suggestion
        unnest_relation.alias(Some(self.new_table_alias("unnset_table_1", vec![alias.to_string()])));
```

Then, we can get the result like
```sql
SELECT "UNNEST(make_array(Int64(1),Int64(2),Int64(3)))" FROM UNNEST([1, 2, 3]) AS unnest_table ("UNNEST(make_array(Int64(1),Int64(2),Int64(3)))")
```

It can not only fit what you want to do but generate a valid SQL. WDYT?

Reply:
https://github.com/apache/datafusion/issues/15233
I created an issue to explain more.